src/internal/log/testing.go

```diff
@@ -0,0 +1,264 @@
+package log
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest"
+	"golang.org/x/exp/maps"
+)
+
+// Test returns a new Context appropriate for use in tests.
+func Test(t testing.TB, opts ...zaptest.LoggerOption) context.Context {
+	l := zaptest.NewLogger(t, opts...)
+	t.Cleanup(zap.ReplaceGlobals(l))
+	t.Cleanup(zap.RedirectStdLog(l))
+	return withLogger(context.Background(), l)
+}
+
+// Test returns a new Context appropriate for use in parallel tests, at the cost of not logging
+// messages sent to the global logger.
+func TestParallel(t testing.TB, opts ...zaptest.LoggerOption) context.Context {
```

Comment:
My understanding is that the preferred way to access this is through `pctx.TestContext`.  Can we add a comment here saying that.

Reply:
That's correct.  I've updated the existing uses of Test/TestParallel to prefer TestContext, and have updated the docs here appropriately.